ksql-engine/src/test/java/io/confluent/ksql/integration/EndToEndIntegrationTest.java

```diff
@@ -0,0 +1,274 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * This test emulates the end to end flow in the quick start guide and ensures that the outputs at each stage
+ * are what we expect. This tests a broad set of KSQL functionality and is a good catch-all.
+ */
+@Category({IntegrationTest.class})
+public class EndToEndIntegrationTest {
+  private static final Logger log = LoggerFactory.getLogger(EndToEndIntegrationTest.class);
+  private IntegrationTestHarness testHarness;
+  private KsqlContext ksqlContext;
+
+  private PageViewDataProvider pageViewDataProvider;
+  private UserDataProvider userDataProvider;
+
+  private final String pageViewTopic = "pageviews";
+  private final String usersTopic = "users";
+
+  private final String pageViewStream = "pageviews_original";
+  private final String userTable = "users_original";
+
+  @Before
+  public void before() throws Exception {
+    testHarness = new IntegrationTestHarness();
+    testHarness.start();
+    Map<String, Object> ksqlConfig = testHarness.ksqlConfig.getKsqlStreamConfigProps();
+    ksqlConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+    ksqlContext = KsqlContext.create(ksqlConfig);
+
+    testHarness.createTopic(pageViewTopic);
+    testHarness.createTopic(usersTopic);
+
+    pageViewDataProvider = new PageViewDataProvider();
+    userDataProvider = new UserDataProvider();
+
+    testHarness.publishTestData(usersTopic, userDataProvider, System.currentTimeMillis() - 10000);
+    testHarness.publishTestData(pageViewTopic, pageViewDataProvider, System.currentTimeMillis());
+
+    ksqlContext.sql(String.format("CREATE TABLE %s (registertime bigint, gender varchar, regionid varchar, " +
+            "userid varchar) WITH (kafka_topic='%s', value_format='JSON');", userTable, usersTopic));
+
+    ksqlContext.sql(String.format("CREATE STREAM %s (viewtime bigint, userid varchar, pageid varchar) " +
+            "WITH (kafka_topic='%s', value_format='JSON');", pageViewStream, pageViewTopic));
+
+  }
+
+  @After
+  public void after() throws Exception {
+    ksqlContext.close();
+    testHarness.stop();
+  }
+
+  @Test
+  public void testKSQLFromEndToEnd() throws Exception {
+    validateSelectFromPageViewsWithSpecificColumn();
+    validateSelectAllFromUsers();
+    String derivedStream = validateSelectAllFromDerivedStream();
+    validateCreateStreamUsingLikeClause(derivedStream);
+  }
+
+  private void validateSelectAllFromUsers() throws Exception {
+    String query = String.format("SELECT * from %s;", userTable);
+    log.debug("Sending query: {}", query);
+
+    List<QueryMetadata> queries = ksqlContext.getKsqlEngine().buildMultipleQueries(false, query, Collections.emptyMap());
```

Comment:
Can we use `ksqlContext.sql(...)` here and then call `ksqlContext.getRunningQueries()`?
Same in the other tests

Reply:
We can't use `ksqlContext.sql` directly because it has an explicit check that the queries submitted through that method are persistent queries. These are not persistent queries, so they need to be submitted directly to the `ksqlEngine`.

Reply:
Seems a bit of a shame. Means that `KsqlContext` is extremely limiting! I know not for this PR, just saying